src/Symfony/Component/Form/Extension/Csrf/EventListener/CsrfValidationListener.php

```diff
@@ -66,15 +66,19 @@ public function preSubmit(FormEvent $event)
         if ($form->isRoot() && $form->getConfig()->getOption('compound') && !$postRequestSizeExceeded) {
             $data = $event->getData();
 
+            if (isset($data[$this->fieldName]) && null !== $data[$this->fieldName] && !\is_string($data[$this->fieldName])) {
```

Comment:
Checking that the value is not `null` is not necessary. Because of the behaviour of `isset()` this code path will never evaluate to `true`.

Reply:
Addressed, thanks!